packages/esm-service-queues-app/src/hooks/useQueueEntries.ts

```diff
@@ -0,0 +1,25 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { type QueueEntry, type QueueEntrySearchCriteria } from '../types';
+import useSWR from 'swr';
+
+export function useQueueEntries(searchCriteria?: QueueEntrySearchCriteria) {
+  const searchParam = new URLSearchParams();
+  for (let [key, value] of Object.entries(searchCriteria)) {
+    searchParam.append(key, value.toString());
+  }
+  searchParam.append('v', 'full');
```

Comment:
I wonder whether it would be useful as a general practice to allow passing in a specific ref to this function (and similar), defaulting to `full` if it is undefined

Reply:
yeah, I like that.

Reply:
That's fine, and I agree that in general you only want to use full when needed.  Isn't there a nice middle ground between ref and full (eg. default?).  (side-note - I have had issues getting default to always work as I expect, has anyone else seen issues with using the default rep?)

Reply:
So basically you are saying don't add any "v=`rep`" at all by default, and only add "v=`rep`" if a `rep` is passed in as a parameter.  Let's pick up the discussion of  the REST api throwing errors in a different thread.

Reply:
`default` works for me, I'll add a `ref = 'default'` param to this funciton